components/hab/src/command/plan/render.rs

```diff
@@ -0,0 +1,176 @@
+            }
+            read_to_string(path.to_string())
+              .expect(&format!("Something went wrong reading: {}", path.to_string()))
+        },
+        // return an empty json block if '--mock-data' isn't defined.
+        // this allows us to merge an empty JSON block
+        None => "{}".to_string(),
+    };
+    // merge mock data into data
+    merge(&mut data, serde_json::from_str(&mock_data).unwrap());
+
+    // create a template renderer
+    let mut renderer = TemplateRenderer::new();
+    // register our template 
+    renderer
+        .register_template_string(&template_path, &template)
+        .expect("Could not register template content");
+    // render our JSON override in our template.
+    let rendered_template = renderer.render(&template_path, &data).ok().unwrap();
+    
+    if print {
+        if !(quiet) {
+          ui.br()?;
+          ui.warn(format!("###======== Rendered template: {}", &template_path))?;
+        }
+
+        println!("{}", rendered_template);
+
+        if !(quiet) {
+          ui.warn(format!("========### End rendered template: {}", &template_path))?;
+        }
+    }
+
+    // if not no render dir (aka "unless no_render_dir == true")
+    if !(no_render_dir) {
+      // Render our template file
+      create_with_template(ui, &format!("{}/{}", render_dir, file_name), &rendered_template, quiet)?;
+    }
+
+    if !(quiet) {
+      ui.br()?;
+    }
+    // not really sure this is correct...
+    Ok(())
+}
+
+fn toml_to_json(cfg: &str) -> Json {
+    // parse TOML string to Value
+    let toml_value = cfg.parse::<Value>().expect("Error parsing TOML");
+    // convert toml to json string
+    let toml_string = serde_json::to_string(&toml_value).expect("Error encoding JSON");
+    // convert to Json::Value
+    serde_json::from_str(&format!(r#"{{ "cfg": {} }}"#, &toml_string)).unwrap()
+}
+
+// merge two Json structs
+fn merge(a: &mut Json, b: Json) {
+    match (a, b) {
+        // not sure I understand this... 
+        (a @ &mut Json::Object(_), Json::Object(b)) => {
+            // not sure I understand why we unwrap this
+            let a = a.as_object_mut().unwrap();
+            // Iterate through key/values in Json object b, 
+            // merge with Json object b
+            for (k, v) in b {
+                merge(a.entry(k).or_insert(Json::Null), v);
+            }
+        }
+        // or this...
+        (a, b) => *a = b,
+    }
+}
+
+// This is almost a dupe of the method in plan/init, except we don't care if the file exists and go
+// ahead and overwite it.  I feel like maybe a different name would be good?
+fn create_with_template(ui: &mut UI, location: &str, template: &str, quiet: bool) -> Result<()> {
+    let path = Path::new(&location);
+    if !(quiet) {
+        ui.status(Status::Creating, format!("file: {}", location))?;
+    }
+    // If the directory doesn't exist we need to make it.
+    if let Some(directory) = path.parent() {
```

Comment:
I don't think you need to check, based on [the docs](http://doc.rust-lang.org/1.32.0/std/fs/fn.create_dir_all.html) (emphasis mine):
> Recursively create a directory and all of its parent components *if they are missing*.

Reply:
ok, so this was largely because of the way I passed `location` in as the full `directory/file_name` then needed to split it apart to create the directory.  I'm not sure it did exactly what I expected, because if that fails, I don't know what would happen... (because `path.parent()` returns an `Option` type?)

in [af8ffc7](https://github.com/qubitrenegade/habitat-1/commit/af8ffc793d4b48f3c21c049c04f20e4a75480f28) I make `file_name` a `std::path::Path` and pass them both in to `create_with_template`.  I think this makes more sense than splitting them apart, smashing them together, then splitting them apart again.

Also, this eliminates the need to clone and unwrap the file name.